ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -604,10 +665,21 @@ class NGraphEncapsulateOp : public OpKernel {
       std::shared_ptr<ng::runtime::Tensor> last_ng_tensor =
           input_caches[i].second;
       void* current_src_ptr = (void*)DMAHelper::base(&ctx->input(i));
-      std::shared_ptr<ng::runtime::Tensor> current_ng_tensor =
-          GetCurrentNgTensor(current_src_ptr, last_src_ptr, last_ng_tensor,
-                             false, ng_exec, op_backend, ng_element_type,
-                             ng_shape);
+      // if m_executable_can_create_tensor, then get current tensor in a
+      // different way
+      // else use GetCurrentNgTensor.
+      // maybe try to shift the current tensor getting code inside
+      // GetCurrentNgTensor
+      std::shared_ptr<ng::runtime::Tensor> current_ng_tensor = nullptr;
+      if (m_executable_can_create_tensor) {
+        current_ng_tensor = inp_group_from_pipeline[i];
+        // TODO: check.
+        current_ng_tensor->set_stale(true);
```

Comment:
For input tensors, set_stale is set to true/false based on a few conditions
https://github.com/tensorflow/ngraph-bridge/blob/951ca465be49bc271ae3b6e130d6d88a0324809f/ngraph_bridge/ngraph_encapsulate_op.cc#L1040
May be you need to add those here or you can also the new code to GetCurrentNgTensor

Reply:
Moved everything to `GetCurrentNgTensor`